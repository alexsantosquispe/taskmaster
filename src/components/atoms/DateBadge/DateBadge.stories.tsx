import type { Meta, StoryObj } from '@storybook/react';

import { DateBadge } from './DateBadge';

const meta = {
  title: 'Atoms/DateBadge',
  component: DateBadge,
  parameters: {
    layout: 'centered'
  }
} satisfies Meta<typeof DateBadge>;

export default meta;

type Story = StoryObj<typeof DateBadge>;

export const Default: Story = () => {
  return <DateBadge date="2025-07-01T10:24:00-04:00" />;
};

Default.args = {};
