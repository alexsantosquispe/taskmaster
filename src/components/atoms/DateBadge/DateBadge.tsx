import { CalendarIcon } from '../../../icons';
import { formatDateToDayMonth } from '../../../utils';

interface DateBadgeProps {
  date: string;
}

export const DateBadge = ({ date }: DateBadgeProps) => {
  return (
    <div className="flex items-center gap-2 py-4 text-neutral-600 dark:text-white/80">
      <CalendarIcon className="size-4" />
      <span>{formatDateToDayMonth(date)}</span>
    </div>
  );
};
