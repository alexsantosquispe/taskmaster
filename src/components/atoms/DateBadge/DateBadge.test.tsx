import { DateBadge } from './DateBadge';
import { render } from '@testing-library/react';

describe('DateBadge', () => {
  it('should render the component correctly', () => {
    const component = render(<DateBadge date="2025-07-05T14:13:00-04:00" />);

    expect(component).toMatchSnapshot();
  });
});
