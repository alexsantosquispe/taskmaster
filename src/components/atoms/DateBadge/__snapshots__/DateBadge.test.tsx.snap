// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`DateBadge should render the component correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex items-center gap-2 text-neutral-600 dark:text-white/80"
      >
        <svg
          class="size-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 2v4"
          />
          <path
            d="M16 2v4"
          />
          <rect
            height="18"
            rx="2"
            width="18"
            x="3"
            y="4"
          />
          <path
            d="M3 10h18"
          />
          <path
            d="M8 14h.01"
          />
          <path
            d="M12 14h.01"
          />
          <path
            d="M16 14h.01"
          />
          <path
            d="M8 18h.01"
          />
          <path
            d="M12 18h.01"
          />
          <path
            d="M16 18h.01"
          />
        </svg>
        <span>
          Jul 5
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex items-center gap-2 text-neutral-600 dark:text-white/80"
    >
      <svg
        class="size-4"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 2v4"
        />
        <path
          d="M16 2v4"
        />
        <rect
          height="18"
          rx="2"
          width="18"
          x="3"
          y="4"
        />
        <path
          d="M3 10h18"
        />
        <path
          d="M8 14h.01"
        />
        <path
          d="M12 14h.01"
        />
        <path
          d="M16 14h.01"
        />
        <path
          d="M8 18h.01"
        />
        <path
          d="M12 18h.01"
        />
        <path
          d="M16 18h.01"
        />
      </svg>
      <span>
        Jul 5
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
